modules/stampcollection.py

```diff
@@ -29,35 +33,31 @@ def reset_stamps(self):
         self.log.info(self.class_name, status="WIPING STAMP RECORDS")
 
         self.utils.clear_votes()
-        self.utils.update_vote(god_id, str(rob_id), self.red_stamp_value)
-
-    def update_vote(self, stamp_type, from_id, to_id, negative=False, recalculate=True):
+        self.update_utils()
+        self.calculate_stamps()
 
-        if to_id == stampy_id:
-            # votes for stampy do nothing
-            return
+    def update_vote(self, emoji: str, from_id: Union[int, str], to_id: Union[int, str],
+                    *, negative: bool = False, recalculate: bool = True):
 
-        if to_id == from_id:
-            # votes for yourself do nothing
+        if (to_id == stampy_id  # votes for stampy do nothing
+            or to_id == from_id # votes for yourself do nothing
+            or emoji not in vote_strengths_per_emoji): # votes with emojis other than stamp and goldstamp do nothing            
             return
-
-        vote_strength = 0
-        if stamp_type == "stamp":
-            vote_strength = self.red_stamp_value
-        elif stamp_type == "goldstamp":
-            vote_strength = self.gold_stamp_value
-
+        
+        vote_strength = vote_strengths_per_emoji[emoji]
         if negative:
-            # if negative is True we are going to subtract the vote value
-            vote_strength = -vote_strength
+            vote_strength *= -1
 
         self.total_votes += vote_strength
         self.utils.update_vote(from_id, to_id, vote_strength)
-        self.utils.users = self.utils.get_users()
-        self.utils.update_ids_list()
+        self.update_utils()
         if recalculate:
             self.calculate_stamps()
 
```

Comment:
I moved updating utils into a separate method so that it's no longer necessary to call `update_vote` just to update utils